pandas/_libs/lib.pyx

```diff
@@ -1362,7 +1372,11 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
 
     elif PyDateTime_Check(val):
         if is_datetime_array(values):
-            return "datetime"
+            if (is_datetimetz_na_array(values)
```

Comment:
you just need is_datetimetz_array

Reply:
Sorry this was quite stupid the part <code>or is_datetime_tz_array(values)</code> would never be executed. I have fixed this.